src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs

```diff
@@ -557,6 +554,43 @@ private bool GetCanVersionIncremental(IncrementalStatus buildInfoIncrementalStat
             return true;
         }
 
+        private IEnumerable<FileItem> GetFilesToCalculateAttributes()
+        {
+            var keys = new HashSet<string>();
+            foreach (var f in _parameters.Files.EnumerateFiles())
+            {
+                var fileKey = ((RelativePath)f.File).GetPathFromWorkingFolder().ToString();
+                keys.Add(fileKey);
+                yield return new FileItem
+                {
+                    PathFromWorkingFolder = fileKey,
+                    FullPath = f.FullPath,
+                    IsFromSource = true,
+                };
+            }
+            if (CurrentBuildVersionInfo.Dependency != null)
+            {
+                foreach (var f in CurrentBuildVersionInfo.Dependency.GetAllDependentNodes())
+                {
+                    var p = RelativePath.TryParse(f);
+                    if (p == null)
```

Comment:
we can only check whether the path contains illegal character. for the case that path doesn't exist. we should still keep it considering the non-existed file might be added in the next run.

---
In reply to: [93843377](https://github.com/dotnet/docfx/pull/1048#discussion_r93843377) [](ancestors = 93843377)